components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/DefaultSessionManager.cs

```diff
@@ -187,22 +191,25 @@ public class DefaultSessionManager : ISessionManager, IDisposable {
         /// stop all sessions
         /// </summary>
         /// <returns></returns>
-        private async Task StopAsync() {
-
-            Try.Op(() => _cts?.Cancel());
+        public async Task StopAsync() {
+            var processingTasks = new List<Task>();
             try {
+                _logger.Information("Stopping all sessions");
+                foreach (var session in _sessions.ToList()) {
```

Comment:
ToList() adds execution time, is it really necessary?

Reply:
yes, the _session is going to change during foreach